packages/aws-arch/src/internal/resources/helpers.ts

```
@@ -0,0 +1,66 @@
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ ******************************************************************************************************************** */
+import { AwsAsset } from "../../generated/assets";
+import { normalizeComparisonString } from "../../utils";
+import { NORMALIZED_RESOURCE_VARIANTS } from "./variants";
+
+/** @internal */
+const RESOURCE_LOOKUP: { [key: string]: string } = Object.fromEntries([
+  ...Object.keys(AwsAsset.Resources).flatMap((key) => {
+    return [
+      // constant self lookup
+      [key, key],
+      // constant comparable
+      [normalizeComparisonString(key), key],
+    ];
+  }),
+  ...Object.entries(NORMALIZED_RESOURCE_VARIANTS).flatMap(([key, variants]) => {
+    if (variants == null) return [];
+    return variants.map((variant) => [normalizeComparisonString(variant), key]);
+  }),
+]);
+
+/**
+ * Resolve resource name value to asset based service key.
+ * @internal
+ */
+export function resolveResourceName(
+  value: string,
+  service?: AwsAsset.Service
+): AwsAsset.Resource {
+  // constant lookup first for perf
+  const comparableValue = normalizeComparisonString(value);
+  if (comparableValue in RESOURCE_LOOKUP) {
+    return RESOURCE_LOOKUP[comparableValue] as AwsAsset.Resource;
+  }
+
+  if (service) {
+    const servicePrefixed = `${normalizeComparisonString(
+      service
+    )}${comparableValue}`;
+    if (servicePrefixed in RESOURCE_LOOKUP) {
+      const resourceName = RESOURCE_LOOKUP[
+        servicePrefixed
+      ] as AwsAsset.Resource;
+      RESOURCE_LOOKUP[comparableValue] = resourceName;
+      return resourceName;
+    }
+  }
+
+  throw new Error(
```

I notice we swallow these errors in consuming code. Does it still make sense to throw or should we log instead?

---

How to handle resource name resolution is up to the consumer, as commented above. Depending on the mapping it is often very common for this to throw an error and therefore there spamming the console does not add value.